### PR TITLE
new package: py-youtube-dl + fixes for dependencies

### PR DESCRIPTION
**Pull request by glennpj**
This PR adds the py-youtube-dl program. In addition, there are a couple
of dependency packages that needed to be updated.

* ffmpeg
This is needed by py-youtube-dl. However, the spack ffmpeg recipe does
not include a lot of options, specifically, a dependency on openssl for
working with the https protocol.

- Added updated version.
- Added variants for the different licensing options.
- Added "meta" variants for X and drawtext. These turn on/off several
  options.
- Set variants and dependencies for many options. The defaults are based
  on the configuration settings in ffmpeg.
- Set dependencies that were missing or that will likely get pulled in
  from the system.

* libxml2
The ffmpeg+libxml2 variant initially failed to build. The issue is that
libxml2 sets the headers property to

include_dir = self.spec.prefix.include.libxml2

The ffmpeg configure looks for prefix.include and fills in the rest.
This could probably be patched in ffmpeg but the headers property in the
libxml2 recipe is not consistent with the environment module or the
pkgconfig file, both of which set the headers path to prefix.include.
This PR sets the libxml2 headers property to

include_dir = self.spec.prefix.include

A spot check of a few libxml2 dependents did not rreveal any problems
with this change.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libxml2/package.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class Libxml2(AutotoolsPackage):
 
     @property
     def headers(self):
-        include_dir = self.spec.prefix.include.libxml2
+        include_dir = self.spec.prefix.include
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libxml2/package.py:43]: I think this change will break other packages.

File: var/spack/repos/builtin/packages/py-youtube-dl/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class PyYoutubeDl(PythonPackage):
+    """Command-line program to download videos from YouTube.com and other video
+    sites."""
+
+    homepage = "https://github.com/ytdl-org/youtube-dl"
+    url      = "https://pypi.io/packages/source/y/youtube_dl/youtube_dl-2020.3.24.tar.gz"
+
+    version('2020.3.24', sha256='4b03efe439f7cae26eba909821d1df00a9a4eb82741cb2e8b78fe29702bd4633')
+
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-youtube-dl/package.py:18]: Are you sure it's needed at run-time?

File: var/spack/repos/builtin/packages/libxml2/package.py (hunk 1 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/libxml2/package.py:43]: The libxml2 dependents that I have on my system build fine with the change. I checked a handful of packages that I do not use and they also built fine. I have not, however, checked every libxml2 dependent. I do know that some of those packages are broken for other reasons. I can try to check them more systematically but that will take a while and I am not sure when I will be able to get to it.

In the meantime I will commit a change to this PR to back out the libxml2 header property change and comment out the libxml2 variant from ffmpeg. The libxml2 option in ffmpeg has limited scope so is likely not too important. I do not want that to hold up this PR.

File: var/spack/repos/builtin/packages/py-youtube-dl/package.py (hunk 2 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-youtube-dl/package.py:18]: Yes, of course, I am sure.

```console
$ module load py-youtube-dl
$ module list py-setuptools

Currently Loaded Modules Matching: py-setuptools
  1) py-setuptools/41.4.0_gcc-9.2.0

$ youtube-dl --version
2020.03.24
```
```console
$ module unload py-setuptools/41.4.0_gcc-9.2.0
$ youtube-dl --version
Traceback (most recent call last):
  File "/opt/ssoft/apps/2020.1/linux-centos7-sandybridge/gcc-9.2.0/py-youtube-dl-2020.3.24-2qh2wka/bin/youtube-dl", line 6, in <module>
    from pkg_resources import load_entry_point
ModuleNotFoundError: No module named 'pkg_resources'
```

**Pull request merged**